src/storage/src/controller.rs

```diff
@@ -1382,8 +1438,8 @@ where
                     .state
                     .exports
                     .get(&export_id)
-                    .expect("Dangling export reference")
-                    .write_frontier,
```

Comment:
IIUC, this is doing the heavy lifting of the PR -- if we've registered an exported collection without a corresponding export, we don't know what as-of it might choose, so we hold back the read capability to the minimum possible. If that sounds right?

Reply:
That's correct!  This is the same way we (before this PR) held back compaction between `create_export` and the sink first reporting back how much it's written out

I'll add a comment before merging for the benefit of whoever reads this next